packages/cloudflare/src/utils/generate-routes-json.ts

```diff
@@ -0,0 +1,267 @@
+import type { AstroConfig, RouteData, RoutePart } from 'astro';
+
+import { existsSync } from 'node:fs';
+import { writeFile } from 'node:fs/promises';
+import { posix } from 'node:path';
+import { fileURLToPath } from 'node:url';
+import glob from 'tiny-glob';
+import { removeLeadingForwardSlash, removeTrailingForwardSlash } from './assets.js';
+
+function prependForwardSlash(path: string) {
+	return path[0] === '/' ? path : `/${path}`;
+}
+
+export const getParts = (part: string) => {
+	const result: RoutePart[] = [];
+	part.split(/\[(.+?\(.+?\)|.+?)\]/).map((str, i) => {
+		if (!str) return;
+		const dynamic = i % 2 === 1;
+
+		const [, content] = dynamic ? /([^(]+)$/.exec(str) || [null, null] : [null, str];
+
+		if (!content || (dynamic && !/^(?:\.\.\.)?[\w$]+$/.test(content))) {
+			throw new Error('Parameter name must match /^[a-zA-Z0-9_$]+$/');
+		}
+
+		result.push({
+			content,
+			dynamic,
+			spread: dynamic && /^\.{3}.+$/.test(content),
+		});
+	});
+
+	return result;
+};
+
+const segmentsToCfSyntax = (segments: RouteData['segments'], _config: AstroConfig) => {
+	const pathSegments = [];
+	if (removeLeadingForwardSlash(removeTrailingForwardSlash(_config.base)).length > 0) {
+		pathSegments.push(removeLeadingForwardSlash(removeTrailingForwardSlash(_config.base)));
+	}
+	for (const segment of segments.flat()) {
+		if (segment.dynamic) pathSegments.push('*');
+		else pathSegments.push(segment.content);
+	}
+	return pathSegments;
+};
+
+class TrieNode {
+	children: Map<string, TrieNode> = new Map();
+	isEndOfPath = false;
+	hasWildcardChild = false;
+}
+
+class PathTrie {
+	root: TrieNode;
+
+	constructor() {
+		this.root = new TrieNode();
+	}
+
+	insert(path: string[]) {
+		let node = this.root;
+		for (const segment of path) {
+			if (segment === '*') {
+				node.hasWildcardChild = true;
+				break;
+			}
+			if (!node.children.has(segment)) {
+				node.children.set(segment, new TrieNode());
+			}
+
+			// biome-ignore lint/style/noNonNullAssertion: <explanation>
+			node = node.children.get(segment)!;
+		}
+
+		node.isEndOfPath = true;
+	}
+
+	private dfs(node: TrieNode, path: string[], allPaths: string[][]): void {
+		if (node.hasWildcardChild) {
+			allPaths.push([...path, '*']);
+			return;
+		}
+
+		if (node.isEndOfPath) {
+			allPaths.push([...path]);
+		}
+
+		for (const [segment, childNode] of node.children) {
+			this.dfs(childNode, [...path, segment], allPaths);
+		}
+	}
+
+	getAllPaths(): string[][] {
+		const allPaths: string[][] = [];
+		this.dfs(this.root, [], allPaths);
+		return allPaths;
+	}
+}
+
+export default async function (
```

Comment:
I would export a named function instead of mixing default exports and named exports. This is usually bad for three-shaking and other stuff.